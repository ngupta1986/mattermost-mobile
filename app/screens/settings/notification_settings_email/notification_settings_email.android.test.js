// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({adapter: new Adapter()});

import {shallowWithIntl} from 'test/intl-test-helper';

import NotificationSettingsMobileAndroid from './notification_settings_email.android.js';

describe('NotificationSettingsMobileAndroid', () => {
    const baseProps = {
        currentUserId: 'current_user_id',
        emailInterval: '30',
        enableEmailBatching: false,
        navigator: {setOnNavigatorEvent: () => {}}, // eslint-disable-line no-empty-function
        actions: {
            savePreferences: () => {}, // eslint-disable-line no-empty-function
        },
        sendEmailNotifications: true,
        siteName: 'Mattermost',
        theme: {
            centerChannelBg: '#aaa',
            centerChannelColor: '#aaa',
        },
    };

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <NotificationSettingsMobileAndroid {...baseProps}/>
        );

        const style = {
            divider: {},
            modal: {},
            modalBody: {},
            modalTitleContainer: {},
            modalTitle: {},
            modalOptionDisabled: {},
            modalHelpText: {},
            modalFooter: {},
            modalFooterContainer: {},
            modalFooterOptionContainer: {},
            modalFooterOption: {},
        };

        expect(wrapper.instance().renderEmailSection()).toMatchSnapshot();
        expect(wrapper.instance().renderEmailNotificationsModal(style)).toMatchSnapshot();
    });

    test('should match state on setEmailNotifications', () => {
        const wrapper = shallowWithIntl(
            <NotificationSettingsMobileAndroid {...baseProps}/>
        );

        wrapper.setState({email: 'false', interval: '0'});
        wrapper.instance().setEmailNotifications('30');
        expect(wrapper.state({email: 'true', interval: '30'}));

        wrapper.instance().setEmailNotifications('0');
        expect(wrapper.state({email: 'false', interval: '0'}));

        wrapper.instance().setEmailNotifications('3600');
        expect(wrapper.state({email: 'true', interval: '3600'}));
    });

    test('should match state on handleClose', () => {
        const wrapper = shallowWithIntl(
            <NotificationSettingsMobileAndroid {...baseProps}/>
        );

        wrapper.setState({showEmailNotificationsModal: true});
        wrapper.instance().handleClose();
        expect(wrapper.state('showEmailNotificationsModal')).toEqual(false);
    });

    test('should saveEmailNotifyProps and handleClose on handleSaveEmailNotification', () => {
        const wrapper = shallowWithIntl(
            <NotificationSettingsMobileAndroid {...baseProps}/>
        );

        const instance = wrapper.instance();
        instance.saveEmailNotifyProps = jest.fn();
        instance.handleClose = jest.fn();

        instance.handleSaveEmailNotification();
        expect(instance.saveEmailNotifyProps).toHaveBeenCalledTimes(1);
        expect(instance.handleClose).toHaveBeenCalledTimes(1);
    });

    test('should match state on showEmailModal', () => {
        const wrapper = shallowWithIntl(
            <NotificationSettingsMobileAndroid {...baseProps}/>
        );

        wrapper.setState({showEmailNotificationsModal: false});
        wrapper.instance().showEmailModal();
        expect(wrapper.state('showEmailNotificationsModal')).toEqual(true);
    });
});
