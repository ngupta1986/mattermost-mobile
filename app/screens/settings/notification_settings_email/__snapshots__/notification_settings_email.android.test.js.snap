// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationSettingsMobileAndroid should match snapshot 1`] = `
<sectionItem
  action={[Function]}
  actionType="default"
  description={
    <InjectIntl(FormattedText)
      defaultMessage="Immediately"
      id="user.settings.notifications.email.immediately"
    />
  }
  label={
    <InjectIntl(FormattedText)
      defaultMessage="Send email notifications"
      id="user.settings.notifications.email.send"
    />
  }
  theme={
    Object {
      "centerChannelBg": "#aaa",
      "centerChannelColor": "#aaa",
    }
  }
/>
`;

exports[`NotificationSettingsMobileAndroid should match snapshot 2`] = `
<Component
  animationType="slide"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={false}
>
  <Component
    style={undefined}
  >
    <Component
      style={Object {}}
    >
      <Component
        style={Object {}}
      >
        <Component
          style={Object {}}
        >
          <InjectIntl(FormattedText)
            defaultMessage="Send email notifications"
            id="user.settings.notifications.email.send"
            style={Object {}}
          />
        </Component>
        <RadioButtonGroup
          name="emailSettings"
          onSelect={[Function]}
          options={
            Array [
              Object {
                "checked": true,
                "label": "Immediately",
                "value": "30",
              },
              Object {
                "checked": false,
                "label": "Never",
                "value": "0",
              },
            ]
          }
        />
        <InjectIntl(FormattedText)
          defaultMessage="Email notifications are sent for mentions and direct messages when you are offline or away from {siteName} for more than 5 minutes."
          id="user.settings.notifications.emailInfo"
          style={Object {}}
          values={
            Object {
              "siteName": "Mattermost",
            }
          }
        />
      </Component>
      <Component
        style={Object {}}
      >
        <Component
          style={Object {}}
        />
        <Component
          style={Object {}}
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={Object {}}
          >
            <InjectIntl(FormattedText)
              defaultMessage="CANCEL"
              id="mobile.notification_settings.modal_cancel"
              style={Object {}}
            />
          </TouchableOpacity>
          <Component>
            <Component
              style={
                Object {
                  "marginRight": 10,
                }
              }
            />
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={Object {}}
            >
              <InjectIntl(FormattedText)
                defaultMessage="SAVE"
                id="mobile.notification_settings.modal_save"
                style={Object {}}
              />
            </TouchableOpacity>
          </Component>
        </Component>
      </Component>
    </Component>
  </Component>
</Component>
`;
